src/components/CampaignImport.vue

```diff
@@ -0,0 +1,89 @@
+<script setup lang="ts">
+import { ref } from "vue";
+import { CampaignStore } from "@/store/CampaignStore";
+import { HeroStore } from "@/store/HeroStore";
+import { useRouter } from "vue-router";
+import BaseModal from "@/components/BaseModal.vue";
+import { XMarkIcon } from "@heroicons/vue/24/solid";
+import { Campaign } from "@/store/Campaign";
+import type { Hero } from "@/store/Hero";
+import { nanoid } from "nanoid";
+
+const isOpen = ref(false);
+const campaignStore = CampaignStore();
+const heroStore = HeroStore();
+const router = useRouter();
+
+const token = ref("");
+
+function importCampaign() {
+  const data = JSON.parse(atob(token.value));
+
+  const campaignId = nanoid();
```

Comment:
const nanoid = customAlphabet("1234567890", 5);
before calling nanoid();

nanoid should be configured so it follows the same rules as the other setup. Should have been extracted to it's own class by me.